open-craft/opencraft
# Adds support for multiple active appservers

Replaces the `OpenEdXInstance.active_appserver` field with a `OpenEdXAppServer.is_active` boolean, allowing more than one active appserver on an instance.

**Testing Instructions**

1. Using the current master branch, create a new instance, and spawn 3 appservers.
1. Activate the first appserver, and wait for the load balancer to be updated.  Ensure that requests are reaching the first appserver.
1. Switch to this branch, and run the migrations.
1. Via the OCIM GUI, ensure that only first appserver is still active.  Also check that the green tick mark is shown for the instance.
1. Activate the second appserver, and check that the GUI updates to add the second active server.
1. Wait for the load balancer to be updated.  Ensure that requests are reaching both appservers.
1. Activate the third appserver, and wait for the load balancer to be updated. 
1. Via the OCIM GUI, ensure that the green tick mark is shown for the instance.
1. Via the OCIM shell, run `instance.terminate_obsolete_appserver(days=0)`.  Ensure that no active appservers are terminated.
1. Via the OCIM GUI, deactivate one of the active appservers.  Ensure that the green tick mark is still shown for the instance.
1. Via the OCIM shell, run `instance.terminate_obsolete_appserver(days=0)`.  Ensure that only the inactive appserver is terminated.
1. Via the OCIM GUI, deactivate all but one of the active appservers.  Ensure that the "Deactivate appserver" button is disabled now that there's only one remaining active appserver.

**Reviewer**
